op-challenger/config/config.go

```diff
@@ -129,26 +129,14 @@ type Config struct {
 	L2Rpc string // L2 RPC Url
 
 	// Specific to the cannon trace provider
-	CannonBin                     string   // Path to the cannon executable to run when generating trace data
-	CannonServer                  string   // Path to the op-program executable that provides the pre-image oracle server
+	Cannon                        vm.Config
 	CannonAbsolutePreState        string   // File to load the absolute pre-state for Cannon traces from
 	CannonAbsolutePreStateBaseURL *url.URL // Base URL to retrieve absolute pre-states for Cannon traces from
```

Comment:
Yeah these are a bit awkward. The actual vm execution always uses a single prestate but it might be a different prestate for different games (each of which has it's own vm.Executor) - hence the `PreStateBaseURL` variant to allow for downloading the required prestate.  So these values actually mean something different to what a prestate in the `vm.Config` would really mean (generic source of prestates for all games vs specific prestate for a particular game).  It would definitely be nice for the `vm.Config` to include the prestate rather than having to pass it in separately though.

I might leave it as-is for this PR but it's worth revisiting and seeing if there's a design that makes sense.